blog/2023-06-20-Deployment-vs-Release.md

```diff
@@ -0,0 +1,56 @@
+
+Deployment and release practices often hit plenty common challenges that can have an impact on their success, especially those practices that have grown organically over time rather than through explicit direction and planning. Recognising these obstacles is the first step towards mitigating them and, towards enhancing the efficiency of these processes. 
+For example, deploying a new version could be a rigid and laborious process. Changes, often require a significant amount of effort to implement, especially when dependencies or configurations are separated from the artefact, managed by different teams or third parties. 
+Then we have the human error factor, brought on by manual deployment processes. The lack of automation can lead to mistakes in configuration, incorrect file transfers, or forgotten/overlooked steps, resulting in failed deployments and service disruptions. This underscores the mantra in modern development: automate everything!
+In deployment models where changes accumulate behind infrequent release events, coordinating all necessary stakeholders from developers and testers, operations ans support, to marketing and leadership - it becomes an overly complex, challenging and unrepeatable process. This usually creates a lack of trust across the teams involved, but an issue that can be alleviated through good practices and open communication.
+Finally, when users discover issues or bugs that slipped past testing, experience unexpected downtime during deployments, or suffer from reduced performance. All these factors can lead to user dissatisfaction.This can create churn in your product and cause loss of trust within your customer.
+Considering all these challenges, it's clear that each of them individually, or even combined (argh!), can considerably impact a product's progress and success. Taking time to discover and understand your bottlenecks, and taking steps to address them by communication, restoring and building trust, and leveraging modern practices and tools to tackle these issues is not just tech debt, but an essential strategy in creating the most value for your organisation.
+
+## Feature Flags
+
+Feature flags, are a simple yet powerful enabler that allow teams to make changes without exposing those changes to the user. This allows teams to deploy whenever they choose, and take steps towards continuous deployment, as any changes, new features or even unfinished code is protected from the user behind the feature flags.
```

Comment:
I would define 'unfinished' code a little more - This could put shivers down your spine